config/main.py

```diff
@@ -1060,7 +1060,8 @@ def load(filename, yes):
 @click.option('-l', '--load-sysinfo', is_flag=True, help='load system default information (mac, portmap etc) first.')
 @click.option('-n', '--no_service_restart', default=False, is_flag=True, help='Do not restart docker services')
 @click.argument('filename', required=False)
-def reload(filename, yes, load_sysinfo, no_service_restart):
+@pass_db
```

Comment:
Most of the CLI handlers have the @click.pass_context decorator .. where we pass the ctx to the command handler. I see here too we store the Db() object in the context ctx.obj = Db() -- So could we pass the pass_context decorator itself ....something like 
ctx.obj['globalDB'] = Db()
and use the @click.pass_context decorator itself ? 

Would this approach also help than creating a separate decorator for db object ? would like to get your opinion. It could be a generic approach to add various objects into the ctx and use it.

Reply:
check out here. the ctx is a linked list, since we created the ctx at the very top level, it could be the case some group will create its own ctx, and we would need to use find_object to find the top level ctx. therefore, click introduce this explicit pass decorator which allow you to get the ctx by object type. 

https://click.palletsprojects.com/en/7.x/complex/?highlight=make_pass_decorator

at any level, you can directly access this ctx.

Reply:
Thanks @lguohan for the pointer for pass_decorator. I had one question though .. on the config command handlers which already have a  ctx being passed  from parent -- will we be able to use this new @pass_db which is nothing but a context created based on object_type. 
eg:
@config.group(cls=AbbreviationGroup)
@click.pass_context
def portchannel(ctx):
    config_db = ConfigDBConnector()
    config_db.connect()
    ctx.obj = {'db': config_db}

Reply:
currently, that code still works, but to unified the logic, we need to refactor that piece of code.